### ISSUE TITLE
Remove `requireCertainty` param from isFromPublicDomain API call

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#5312]
## Problem
Coming from [this thread](https://expensify.slack.com/archives/C020EPP9B9A/p1631765876317300?thread_ts=1631749275.291700&cid=C020EPP9B9A), we currently check public domains with `requireCertainty=true`, which throws a Clearbit exception if the domain is not found in the common domain list or bedrock cache. In the meantime, we assume that users have a public domain. 

## Why is this important?
This affects their experience, because they are not able to do some things on a public domain and we could potentially lose them. 

## Solution
Given that we already queue a bedrock job for domains that are not found, I think we should remove the [requireCertainty](https://github.com/Expensify/App/blob/574aa7525a311f51c06695e9c14dcecba12e9869/src/libs/API.js#L752) param as a default from requests. This means that we would assume a private domain while we check Clearbit and unblock users. 

We might also have to add that param to calls where we actually need certainty like the KYC flow.

cc @roryabraham @flodnv @iwiznia because of the discussions [here](https://github.com/Expensify/Expensify/issues/125592) and [here](https://github.com/Expensify/Expensify/issues/120480).

**Comment by iwiznia** [comment]
Can you give me a bit more context on what exactly we use this information for and what it is blocking users from doing? It seems that we call this right after user logs in and that we retry it (to get the updated data) when it fails.

**Comment by luacmartins** [comment]
I looked into this a bit more and here's what I found. 

1. We call `User_isFromPublicDomain` whenever a user logs in or refreshes the app.
2. We are only using this information when a user tries to add a verified bank account `Settings > Workspace > Expensify Card > Get started`.
3. The API call will throw an exception when we need to reach out to Clearbit for certainty and we try again after 10 minutes. In the meantime, the Onyx key will be `undefined`, but we default to a [private domain](https://github.com/Expensify/App/blob/837ada35ea6c714f5e2b2895fbc039fa30def954/src/pages/workspace/WorkspaceCardPage.js#L63) anyways.
 
To my understanding, adding a verified bank account is a critical flow where we actually want private domain certainty. Is that the case? If so, we are actually not enforcing it and should start doing so. If not, we could remove the `requireCertainty=true` param and avoid throwing the exception and retrying the API call.

Thoughts?
